Source/SwiftyDropbox/Shared/Handwritten/Request+TokenRefresh.swift

```diff
@@ -0,0 +1,142 @@
+///
+/// Copyright (c) 2020 Dropbox, Inc. All rights reserved.
+///
+
+import Foundation
+import Alamofire
+
+/// Completion handler for ApiRequest.
+enum RequestCompletionHandler {
+    case dataCompletionHandler((DefaultDataResponse) -> Void)
+    case downloadFileCompletionHandler((DefaultDownloadResponse) -> Void)
+}
+
+/// Protocol defining an API request object.
+protocol ApiRequest {
+    @discardableResult func progress(handler: @escaping Alamofire.Request.ProgressHandler) -> Self
+
+    @discardableResult
+    func response(queue: DispatchQueue?, completionHandler: RequestCompletionHandler) -> Self
```

Comment:
I had to look at these methods implementation to see what they were doing. These should be named:

```
setProgressHandler(_ handler: @escaping Alamofire.Request.ProgressHandler) -> Self)
```
and
```
setCompletionHandler(_ handler: RequestCompletionHandler, onQueue: DispatchQueue?) -> Self
```

The current naming is very confusing.

Reply:
yeah, the names are not good, updated.